polaris.shopify.com/src/components/ComponentExamples/ComponentExamples.tsx

```diff
@@ -81,61 +85,172 @@ const ComponentExamples = ({examples}: Props) => {
+        setSelectedTab(0);
+        break;
+
+      case 'End':
+        setSelectedTab(examplesLength);
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  const buttonRefs = useRef<(HTMLButtonElement | null)[]>([]);
+
   useEffect(() => {
-    setSelectedIndex(0);
+    buttonRefs.current = buttonRefs.current.slice(0, examples.length);
   }, [examples]);
 
-  return (
-    <Tab.Group
-      defaultIndex={0}
-      selectedIndex={selectedIndex}
-      onChange={setSelectedIndex}
-    >
-      <Tab.List>
-        <div className={styles.ExamplesList} id="examples">
-          {examples.map((example) => {
-            return (
-              <Tab key={example.fileName}>
-                <span>{example.title}</span>
-              </Tab>
-            );
-          })}
-        </div>
-      </Tab.List>
-
-      <Tab.Panels>
-        {examples.map(({fileName, description, code}) => {
-          const exampleUrl = `/examples/${fileName.replace('.tsx', '')}`;
-
-          return (
-            <Tab.Panel key={fileName}>
-              {description ? <Markdown {...description} /> : null}
-              <div className={styles.ExampleFrame}>
-                <iframe
-                  src={exampleUrl}
-                  height={iframeHeight}
-                  onLoad={handleExampleLoad}
-                  id={exampleIframeId}
-                />
-                <div className={className(styles.Buttons, 'light-mode')}>
-                  <CodesandboxButton
-                    className={styles.CodesandboxButton}
-                    code={code}
-                  />
-                </div>
-              </div>
-
-              <Code
-                code={[
-                  {title: 'React', code: code.trim()},
-                  {title: 'HTML', code: htmlCode},
-                ]}
+  useEffect(() => {
+    const currentButtonRef = buttonRefs.current[exampleIndex];
+    if (currentButtonRef) {
+      currentButtonRef.focus();
+    }
+  }, [exampleIndex]);
+
+  const tablistId = 'component-examples-tablist';
+
+  const tabButtons = examples.map((example, index) => {
+    return (
+      <button
+        role="tab"
+        type="button"
+        id={`tab-${index}`}
+        key={example.fileName}
+        onClick={() => updateSelectedExample(index)}
+        aria-controls={`tabpanel-${index}`}
+        tabIndex={index === exampleIndex ? 0 : -1}
+        ref={(el) => (buttonRefs.current[index] = el)}
+        aria-selected={index === exampleIndex}
+      >
+        <span>{example.title}</span>
+      </button>
+    );
+  });
+
+  const examplesMarkup = examples.map(
+    ({fileName, description, code}, index) => {
+      const exampleUrl = `/examples/${fileName.replace('.tsx', '')}`;
+      const isSelected = index === exampleIndex ? true : false;
+      const classes = classNames(
+        styles.Tab,
+        isSelected ? styles.selected : undefined,
+      );
+
+      return (
+        <div
+          id={`tabpanel-${index}`}
```

Comment:
Just adding new `classNames`, aria labels, and other attributes to this. The rest of this component was untouched